lightly/data/dataset.py

```diff
@@ -216,6 +216,34 @@ def from_torch_dataset(cls, dataset, transform=None, index_to_filename=None):
         dataset_obj.dataset = dataset
         return dataset_obj
 
+    @classmethod
+    def from_hf_dataset(cls, dataset, transform=None, key: str):
+        """Builds a LightlyDataset from a HuggingFace dataset.
+
+        Args:
+            dataset:
+                HuggingFace dataset
+            transform:
+                Image transforms (as in torchvision).
+            key:
+                The key containing images in the dataset.
+
+        Returns:
+            A LightlyDataset object.
+
+        """ 
+        def apply_transform(batch, transform=tranform, key=key):
+            assert key in batch.keys(), f"the provided key, {key} does not exist in the dataset"
+
+            batch[key] = [transform(image) for image in batch[key]]
+
+            return batch
```

Comment:
Not sure if this works with distributed training as local functions cannot be pickled and sent to other processes. We might have to define a transform class that takes the transform and key as input.